File: Algorithms Advanced/others/C++/WeightMatrix.cpp
#include <iostream>
#include <climits>
using namespace std;

int main() 
{
    const int INF = INT_MAX; // Infinity for no edge
    int vertices = 3;
    int weightMatrix[3][3] = {
        {0, 4, INF},
        {INF, 0, 3},
        {INF, INF, 0}
    };

    // Print weight matrix
    for (int i = 0; i < vertices; i++) 
    {
        for (int j = 0; j < vertices; j++) 
        {
            if (weightMatrix[i][j] == INF) cout << "INF ";
            else cout << weightMatrix[i][j] << " ";
        }
        cout << endl;
    }

    return 0;
}

File: Algorithms Advanced/others/Python/WeightMatrix.py
# 4-node weighted graph weight matrix
n = 4
weight_matrix = [[float('inf')] * n for _ in range(n)]

# Add weights
weight_matrix[0][1] = 5  # Weight from node 0 to node 1
weight_matrix[1][2] = 2  # Weight from node 1 to node 2

# Set diagonal to 0 (no cost to self)
for i in range(n):
    weight_matrix[i][i] = 0

# Display
for row in weight_matrix:
    print(row)
